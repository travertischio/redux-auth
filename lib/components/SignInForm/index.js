"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactIntl = require("react-intl");

var _immutable = require("redux-form/immutable");

var _reduxForm = _interopRequireDefault(require("react-form-fields/lib/EmailField/reduxForm"));

var _reduxForm2 = _interopRequireDefault(require("react-form-fields/lib/PasswordField/reduxForm"));

var _required = require("validators/lib/required");

var _email = require("validators/lib/email");

var _field = _interopRequireDefault(require("../Captcha/field"));

var _messages = _interopRequireDefault(require("./messages"));

/**
*
* SignInForm
*
*/
var SignInForm = function SignInForm(props) {
  var captchaRequired = props.captchaRequired,
      handleSubmit = props.handleSubmit,
      pristine = props.pristine,
      submitting = props.submitting;
  var formatMessage = props.intl.formatMessage;
  var emailLabel = formatMessage(_messages["default"].email);
  var passwordLabel = formatMessage(_messages["default"].password);

  var onSubmit = function onSubmit(event) {
    if (event && event.preventDefault) {
      event.preventDefault();
    }

    handleSubmit();
  };

  return _react["default"].createElement("form", {
    onSubmit: onSubmit,
    noValidate: true
  }, _react["default"].createElement(_immutable.Field, {
    name: "email",
    id: "email",
    label: emailLabel,
    placeholder: emailLabel,
    validate: [_required.required, _email.email],
    component: _reduxForm["default"]
  }), _react["default"].createElement(_immutable.Field, {
    name: "password",
    id: "password",
    label: passwordLabel,
    placeholder: passwordLabel,
    validate: [_required.required],
    component: _reduxForm2["default"]
  }), captchaRequired && _react["default"].createElement(_immutable.Field, {
    name: "captcha",
    validate: [_required.required],
    component: _field["default"]
  }), _react["default"].createElement("div", null, _react["default"].createElement("button", {
    type: "submit",
    disabled: pristine || submitting
  }, "Sign In")));
};

SignInForm.defaultProps = {
  captchaRequired: false
};

var _default = (0, _immutable.reduxForm)({
  form: 'signInForm'
})((0, _reactIntl.injectIntl)(SignInForm));

exports["default"] = _default;