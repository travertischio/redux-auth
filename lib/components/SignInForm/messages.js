"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _reactIntl = require("react-intl");

/*
 * SignInForm Messages
 *
 * This contains all the text for the SignInForm component.
 */
var _default = (0, _reactIntl.defineMessages)({
  email: {
    id: 'app.components.SignInForm.email',
    defaultMessage: 'E-mail'
  },
  password: {
    id: 'app.components.SignInForm.password',
    defaultMessage: 'Password'
  }
});

exports["default"] = _default;