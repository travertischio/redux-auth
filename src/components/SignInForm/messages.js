/*
 * SignInForm Messages
 *
 * This contains all the text for the SignInForm component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  email: {
    id: 'app.components.SignInForm.email',
    defaultMessage: 'E-mail',
  },
  password: {
    id: 'app.components.SignInForm.password',
    defaultMessage: 'Password',
  },
});
