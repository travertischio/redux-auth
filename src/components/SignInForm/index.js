/**
*
* SignInForm
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import {
  Field,
  reduxForm,
} from 'redux-form/immutable';
import EmailField from 'react-form-fields/lib/EmailField/reduxForm';
import PasswordField from 'react-form-fields/lib/PasswordField/reduxForm';
import { required as requiredValidator } from 'validators/lib/required';
import { email as emailValidator } from 'validators/lib/email';
import CaptchaField from '~/components/Captcha/field';
import messages from './messages';

const SignInForm = (props) => {
  const {
    captchaRequired,
    handleSubmit,
    pristine,
    submitting,
  } = props;
  const { formatMessage } = props.intl;
  const emailLabel = formatMessage(messages.email);
  const passwordLabel = formatMessage(messages.password);

  const onSubmit = (event) => {
    if (event && event.preventDefault) {
      event.preventDefault();
    }

    handleSubmit();
  };

  return (
    <form onSubmit={onSubmit} noValidate>
      <Field
        name="email"
        id="email"
        label={emailLabel}
        placeholder={emailLabel}
        validate={[requiredValidator, emailValidator]}
        component={EmailField}
      />
      <Field
        name="password"
        id="password"
        label={passwordLabel}
        placeholder={passwordLabel}
        validate={[requiredValidator]}
        component={PasswordField}
      />
      {captchaRequired
        && (
          <Field
            name="captcha"
            validate={[requiredValidator]}
            component={CaptchaField}
          />
        )}
      <div>
        <button
          type="submit"
          disabled={pristine || submitting}
        >
          Sign In
        </button>
      </div>
    </form>
  );
};

SignInForm.propTypes = {
  captchaRequired: PropTypes.bool,
  handleSubmit: PropTypes.func.isRequired,
  intl: PropTypes.object.isRequired,
  pristine: PropTypes.bool.isRequired,
  submitting: PropTypes.bool.isRequired,
};

SignInForm.defaultProps = {
  captchaRequired: false,
};

export default reduxForm({
  form: 'signInForm',
})(injectIntl(SignInForm));
