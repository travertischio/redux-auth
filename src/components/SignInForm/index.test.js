import React from 'react';
import { createComponentWithIntl } from 'react-unit-testing-utils';

import SignInForm from '.';

describe('<SignInForm />', () => {
  it('should render SignInForm', () => {
    const initialState = {};
    const { component } = createComponentWithIntl(<SignInForm />, initialState);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render SignInForm with captcha when has captchaRequired prop', () => {
    const initialState = {};
    const { component } = createComponentWithIntl(<SignInForm captchaRequired />, initialState);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
