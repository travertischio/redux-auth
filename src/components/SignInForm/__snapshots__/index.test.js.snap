// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInForm /> should render SignInForm 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className=""
  >
    <div
      className="form-item"
    >
      <label
        htmlFor="email"
      >
        E-mail
      </label>
      <input
        className="textfield"
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="E-mail"
        type="email"
        value=""
      />
    </div>
  </div>
  <div
    className=""
  >
    <div
      className="form-item"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="textfield"
        id="password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
  </div>
  <div>
    <button
      disabled={true}
      type="submit"
    >
      Sign In
    </button>
  </div>
</form>
`;

exports[`<SignInForm /> should render SignInForm with captcha when has captchaRequired prop 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className=""
  >
    <div
      className="form-item"
    >
      <label
        htmlFor="email"
      >
        E-mail
      </label>
      <input
        className="textfield"
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="E-mail"
        type="email"
        value=""
      />
    </div>
  </div>
  <div
    className=""
  >
    <div
      className="form-item"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="textfield"
        id="password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
  </div>
  <div
    className=""
  >
    <div
      className="g-recaptcha"
      data-badge="bottomright"
      data-size="normal"
      data-tabindex="0"
      data-theme="light"
      data-type="image"
      id="g-recaptcha"
    />
  </div>
  <div>
    <button
      disabled={true}
      type="submit"
    >
      Sign In
    </button>
  </div>
</form>
`;
